src/main/java/net/sf/jabref/gui/openoffice/OpenOfficePanel.java

```diff
@@ -669,6 +674,57 @@ private void pushEntries(boolean inParenthesisIn, boolean withText, boolean addP
 
     }
 
+    /**
+     * Check that all entries in the list have BibTeX keys, if not ask if they should be generated
+     *
+     * @param entries A list of entries to be checked
+     * @return true if all entries have BibTeX keys, if it so may be after generating them
+     */
+    private boolean checkThatEntriesHaveKeys(List<BibEntry> entries) {
+        // Check if there are empty keys
+        boolean emptyKeys = false;
+        for (BibEntry entry : entries) {
+            if (entry.getCiteKey() == null) {
+                // Found one, no need to look further for now
+                emptyKeys = true;
+                break;
+            }
+        }
+
+        // If no empty keys, return true
+        if (!emptyKeys) {
+            return true;
+        }
+
+        // Ask if keys should be generated
+        int answer = JOptionPane.showConfirmDialog(this.frame,
+                Localization.lang("Cannot cite entries without BibTeX keys. Generate keys now?"),
+                Localization.lang("Cite"), JOptionPane.YES_NO_OPTION);
```

Comment:
Dialogs asking a question and then having yes/no  actions are generally not a good idea, because the "thinking overhead" for the user in the moment is too much.  http://ux.stackexchange.com/questions/756/what-are-some-alternatives-to-the-phrase-are-you-sure-you-want-to-xyz-in-confi

Better add a button like  "Generate keys" and a Cancel button.